indexing-service/src/main/java/org/apache/druid/indexing/overlord/config/TaskLockConfig.java

```diff
@@ -36,6 +36,9 @@ public class TaskLockConfig
   @JsonProperty
   private long batchAllocationWaitTime = 0L;
 
+  @JsonProperty
+  private boolean batchAllocationReduceMetadataIO = false;
```

Comment:
Should this be always on and then we remove this config in next release ?

Reply:
Yeah, I guess the changes are safe enough. We can always keep it on. Let me try to run ITs with it turned on.

Reply:
Flag has been set to true by default.